bucket_map/src/bucket_map.rs

```diff
@@ -23,15 +23,15 @@ pub struct BucketMapKeyValue<T> {
 
 #[derive(Debug, Default)]
 pub struct BucketMapConfig {
-    pub num_buckets: usize,
```

Comment:
what is your thinking on max vs num here? That possibly not all buckets are allocated?